codespell_lib/tests/test_dictionary.py

```diff
@@ -57,40 +60,42 @@ def test_dictionary_formatting(fname, in_aspell):
             err = err.lower()
             rep = rep.rstrip('\n')
             try:
-                _check_err_rep(err, rep, in_aspell, fname)
+                _check_err_rep(err, rep, in_aspell, fname, in_dictionary)
             except AssertionError as exp:
                 errors.append(str(exp).split('\n')[0])
     if len(errors):
         raise AssertionError('\n' + '\n'.join(errors))
 
 
-def _check_aspell(phrase, msg, in_aspell, fname):
-    if speller is None:
+def _check_aspell(phrase, msg, in_aspell, fname, languages):
+    if not spellers:  # if no spellcheckers exist
         return  # cannot check
     if in_aspell is None:
         return  # don't check
     if ' ' in phrase:
         for word in phrase.split():
-            _check_aspell(word, msg, in_aspell, fname)
+            _check_aspell(word, msg, in_aspell, fname, languages)
         return  # stop normal checking as we've done each word above
-    this_in_aspell = speller.check(
-        phrase.encode(speller.ConfigKeys()['encoding'][1]))
-    end = 'be in aspell for dictionary %s' % (fname,)
+    this_in_aspell = any(spellers[lang].check(phrase.encode(
+        spellers[lang].ConfigKeys()['encoding'][1])) for lang in languages)
+    end = 'be in aspell dictionaries (%s) for dictionary %s' % (
+        ', '.join(languages), fname,)
     if in_aspell:  # should be an error in aspell
         assert this_in_aspell, '%s should %s' % (msg, end)
     else:  # shouldn't be
         assert not this_in_aspell, '%s should not %s' % (msg, end)
+        return this_in_aspell
```

Comment:
Not sure why you've added the `return` to the `else` branch but not the `if` branch?

Reply:
ups. I wanted to test something for #1711 and accidently commited it.